pkg/app/kafka.go

```diff
@@ -260,36 +282,79 @@ func (kc *KafkaCluster) Initialize(ctx context.Context) error {
 	return nil
 }
 
-// to produce messages to kafka
-func produce(ctx context.Context, topic string, namespace string) error {
-	partition := 0
-	forwarder, err := K8SServicePortForward(ctx, bootstrapServerHost, namespace, bootstrapServerPort)
+type Message struct {
+	Topic     string `json:"topic"`
+	Key       string `json:"key"`
+	Value     string `json:"value"`
+	Partition int    `json:"partition"`
+	Offset    int    `json:"offset"`
+}
+
+type InsertPayload struct {
+	Records []Records `json:"records"`
+}
+
+type Records struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type Payload struct {
+	Name                     string `json:"name"`
+	AutoOffsetReset          string `json:"auto.offset.reset"`
+	Format                   string `json:"format"`
+	EnableAutoCommit         bool   `json:"enable.auto.commit"`
+	FetchMinBytes            int    `json:"fetch.min.bytes"`
+	ConsumerRequestTimeoutMs int    `json:"consumer.request.timeout.ms"`
+}
+
+type SubscriptionPayload struct {
+	Topics []string `json:"topics"`
+}
+
+// InsertRecord creates a topic and insert a record
+func InsertRecord(ctx context.Context, namespace string) error {
```

Comment:
Let's implement this method on `KafkaCluster`
```suggestion
func (kc *KafkaCluster) InsertRecord(ctx context.Context, namespace string) error {
```